packages/react-reconciler/src/ReactFiberThrow.new.js

```diff
@@ -487,20 +488,24 @@ function throwException(
   } else {
     // This is a regular error, not a Suspense wakeable.
     if (getIsHydrating() && sourceFiber.mode & ConcurrentMode) {
+      const suspenseBoundary = getNearestSuspenseBoundaryToCapture(returnFiber);
       // If the error was thrown during hydration, we may be able to recover by
       // discarding the dehydrated content and switching to a client render.
       // Instead of surfacing the error, find the nearest Suspense boundary
       // and render it again without hydration.
-      const suspenseBoundary = markNearestSuspenseBoundaryShouldCapture(
-        returnFiber,
-        sourceFiber,
-        root,
-        rootRenderLanes,
-      );
       if (suspenseBoundary !== null) {
-        // Set a flag to indicate that we should try rendering the normal
-        // children again, not the fallback.
-        suspenseBoundary.flags |= ForceClientRender;
+        if ((suspenseBoundary.flags & ShouldCapture) === NoFlags) {
+          // Set a flag to indicate that we should try rendering the normal
+          // children again, not the fallback.
+          suspenseBoundary.flags |= ForceClientRender;
```

Comment:
Need to handle the reverse case, too: something throws an error during hydration, then later it suspends. When that happens we should remove the ForceClientRender flag.